lumicks/pylake/force_calibration/tests/data/simulate_ideal.py

```diff
@@ -0,0 +1,194 @@
+    bead_position_contribution = amp_bead_position * np.sin(omega_stage * (time - t_lag)) * 1e-9
+
+    return stage_position, bead_position_contribution
+
+
+def apply_diode_filter(positions, diode_alpha, diode_frequency, time_step):
+    """Applies the simple diode filter model to the data.
+
+    Parameters
+    ----------
+    positions : array_like
+        List of bead positions [m]
+    diode_alpha : float
+        Fraction of the diode response that is instantaneous [-]
+    diode_frequency : float
+        Corner frequency of the filtering effect of the PSD [Hz]
+    time_step : float
+        Time step of the simulation [s]
+    """
+    assert 0 <= diode_alpha <= 1, (
+        "Invalid diode fraction (alpha) provided. " "Should be between 0 and 1"
+    )
+    filter_coeff = np.exp(-2.0 * np.pi * time_step * diode_frequency)
+    filtered = scipy.signal.lfilter([(1 - filter_coeff)], [1, -filter_coeff], positions)
+    return positions * diode_alpha + filtered * (1.0 - diode_alpha)
+
+
+def simulate_calibration_data(
+    duration,
+    sample_rate,
+    bead_diameter,
+    stiffness,
+    viscosity,
+    temperature,
+    pos_response_um_volt,
+    anti_aliasing,
+    oversampling=8,
+    active_sinusoid=None,
+    diode=None,
+):
+    """Simulate Brownian motion in optical trap
+
+    This function simulates the Langevin equation that arises from a bead in an optical trap while
+    the stage is oscillating. Rather than using a basic Euler scheme, we use the more exact
+    simulation method from [1] which does not suffer from discretization error. Equation (72)
+    from [1] reads:
+
+        x[n+1] = c * x[n] + dx
+
+    with:
+
+        c = exp(-stiffness * dt / gamma)
+        dx = sqrt( (1 - c^2) * diffusion * gamma / stiffness ) * normal(0, 1)
+
+    For performance, we recast this simulation as a digital filter applied to a list of values drawn
+    from a normal distribution rather than simulate it with a for loop.
+
+    [1] Nørrelykke, S. F., & Flyvbjerg, H. (2011). Harmonic oscillator in heat bath: Exact
+    simulation of time-lapse-recorded data and exact analytical benchmark statistics. Physical
+    Review E, 83(4), 041103.
+
+    Parameters
+    ----------
+    duration : float
+        Time [s]
+    sample_rate : int
+        Sampling rate [Hz]
+    bead_diameter : float
+        Bead diameter [micron]
+    stiffness : float
+        Spring constant of the trap [pN/nm]
+    viscosity : float
+        Viscosity [Pa*s]
+    temperature : float
+        Temperature [C]
+    pos_response_um_volt : float
+        Response [um/V], also denoted in papers as Rd
+    anti_aliasing : bool
+        Should we anti-alias the data?
+    oversampling : int
+        Oversampling factor. Relatively high oversampling ratios are required to reject aliasing
+        effectively.
+    active_sinusoid : tuple(float, float) or None
+        Parameters for the driving input.
```

Comment:
We end up calling these `driving_amplitude` and `driving_frequency`, so would it make sense to call this: `driving_sinusoid`? Or is this named already consistently used in the literature?

Reply:
I'll rename it to `driving_sinusoid` for consistency. Literature usually just refers to it as driving input.